seqr/migrations/0070_remove_rnasample_dataset_type_and_more.py

```diff
@@ -0,0 +1,123 @@
+    if no_data_samples:
+        for data_type, substring in [('S', 'fraser'), ('T', 'tpm'), ('E', 'outrider')]:
+            count = _update_sample_data_type(no_data_samples.filter(data_source__icontains=substring), data_type)
+            print(f'Inferred data type for {count} inactive {substring} samples')
+
+    # Update primary data type
+    num_splice = _update_sample_data_type(rna_samples.filter(id__in=splice_outlier_samples), 'S')
+    num_tpm = _update_sample_data_type(rna_samples.filter(id__in=tpm_outlier_samples-splice_outlier_samples), 'T')
+    num_expr = _update_sample_data_type(
+        rna_samples.filter(id__in=expression_outlier_samples-tpm_outlier_samples-splice_outlier_samples), 'E')
+    if num_splice or num_tpm or num_expr:
+        print(f'Updated primary data type for {num_splice} splice, {num_expr} expression, and {num_tpm} tpm samples')
+
+    # Add sample models for samples with multiple data types
+    _create_data_type_samples(
+        rna_samples, RnaSeqTpm.objects.using(db_alias), 'T', sample_ids=splice_outlier_samples & tpm_outlier_samples,
+    )
+    _create_data_type_samples(
+        rna_samples,  RnaSeqOutlier.objects.using(db_alias), 'E',
+        sample_ids=expression_outlier_samples & (splice_outlier_samples | tpm_outlier_samples),
+    )
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('seqr', '0069_remove_sample_dataset_type_and_more'),
+    ]
+
+    operations = [
+        migrations.RenameModel('Sample', 'RnaSample'),
+        migrations.RenameModel('NonRnaSample', 'Sample'),
+        migrations.RemoveField(
+            model_name='rnasample',
+            name='dataset_type',
+        ),
+        migrations.RemoveField(
+            model_name='rnasample',
+            name='elasticsearch_index',
+        ),
+        migrations.RemoveField(
+            model_name='rnasample',
+            name='loaded_date',
+        ),
+        migrations.RemoveField(
+            model_name='rnasample',
+            name='sample_id',
+        ),
+        migrations.RemoveField(
+            model_name='rnasample',
+            name='sample_type',
+        ),
+        migrations.AddField(
+            model_name='rnasample',
+            name='data_type',
+            field=models.CharField(choices=[('T', 'TPM'), ('E', 'Expression Outlier'), ('S', 'Splice Outlier')], default='X', max_length=1),
+            preserve_default=False,
+        ),
+        migrations.AlterField(
+            model_name='rnasample',
+            name='data_source',
+            field=models.TextField(),
+        ),
+        migrations.AlterField(
+            model_name='rnasample',
+            name='tissue_type',
+            field=models.CharField(choices=[('WB', 'whole_blood'), ('F', 'fibroblasts'), ('M', 'muscle'), ('L', 'lymphocytes'), ('A', 'airway_cultured_epithelium')], max_length=2),
+        ),
+        migrations.RunPython(add_data_type_rna_samples, reverse_code=migrations.RunPython.noop),
```

Comment:
why make the first migration reversible but not this one?

Reply:
This migration is reversible, this line is just saying that there is no custom python to run in the reverse migration. Thats because `add_data_type_rna_samples` creates new sample models where necessary, but if in the reverse migration we keep those new models no harm happens, and then if we re-ran the forward migration it would not recreate them new models, it would just detect that there were no instances of multiple data types sharing a model. 
In the previous migration, if we did not write reverse code than all the non-RNA models would be deleted, which would be an irreverisible change. Here, the change is reversible with no custom handling